resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -127,6 +121,11 @@ private[spark] class Client(
             .pods()
             .withLabels(driverKubernetesSelectors)
```

Comment:
Currently our PodWatcher watches the driver pod to determine its liveness for uploading files to it.  What we more directly depend on those is the service -- when the readiness probe goes to success and the pod gets added to the service, that seems like the more specific trigger.

Proposal: what if we change the `DriverPodWatcher` to watch for the service gaining a backing pod, instead of the driver pod being ready?  That more directly monitors for the condition we need to submit the job.